3p/facebook.js

```diff
@@ -207,23 +196,40 @@ function getLikeContainer(global, data) {
 }
 
 /**
+ * Create DOM element for the Facebook embedded content plugin.
  * @param {!Window} global
- * @param {!Object} data
+ * @param {!Object} data The element data
+ * @param {string} embedAs
+ * @return {!Element} div
  */
-export function facebook(global, data) {
-  const extension = global.context.tagName;
-  let container;
-
-  if (extension === 'AMP-FACEBOOK-PAGE') {
-    container = getPageContainer(global, data);
-  } else if (extension === 'AMP-FACEBOOK-LIKE') {
-    container = getLikeContainer(global, data);
-  } else if (extension === 'AMP-FACEBOOK-COMMENTS') {
-    container = getCommentsContainer(global, data);
-  } /*AMP-FACEBOOK */ else {
-    container = getEmbedContainer(global, data);
+function getEmbedContainer(global, data, embedAs) {
+  devAssert(isEnumValue(FacebookEmbedType, embedAs));
+  switch (embedAs) {
+    case FacebookEmbedType.PAGE:
+      return getPageContainer(global, data);
```

Comment:
How do these implementations vary? Could it be possible to pass the type downstream to the same iframe instead?

Reply:
They seem pretty similar at a glance. I didn't want to change too much at once with on PR so thought this would be an appropriate boundary for the refactor. Happy to take another pass simplifying things after this PR is merged.